server/src/main/java/org/apache/druid/server/QueryResultPusher.java

```diff
@@ -223,6 +235,8 @@ private Response handleDruidException(ResultsWriter resultsWriter, DruidExceptio
         // also throwing the exception body into the response to make it easier for the client to choke if it manages
         // to parse a meaningful object out, but that's potentially an API change so we leave that as an exercise for
         // the future.
+        trailerFields.put(QueryResource.ERROR_MESSAGE_TRAILER_HEADER,
+            String.format(Locale.ENGLISH, "StatusCode: %s, Category: %s", e.getStatusCode(), e.getCategory()));
```

Comment:
maybe we should also add the message here?

Reply:
Done